test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/FlexibleSyncConfigurationTests.kt

```diff
@@ -0,0 +1,177 @@
+        val config1: SyncConfiguration = SyncConfiguration.with(user1, setOf())
+        val config2: SyncConfiguration = SyncConfiguration.with(user2, setOf())
+        assertNotEquals(config1.hashCode(), config2.hashCode())
+    }
+
+    @Test
+    fun toString_nonEmpty() {
+        val user: User = createTestUser()
+        val config: SyncConfiguration = SyncConfiguration.with(user, setOf())
+        // TODO Currently we use default implementation for `toString()`. This is
+        //  different compared to Realm Java, but it is unclear if this is something
+        //  that is worth implementing?
+        assertTrue(config.toString().startsWith("io.realm.mongodb.internal.SyncConfigurationImpl@"), config.toString())
+    }
+
+    // @Test
+    // fun getPartitionValueThrows() {
+    //     val user: User = createTestUser(app)
+    //     val config: SyncConfiguration = SyncConfiguration.defaultConfig(user)
+    //     assertFailsWith<IllegalStateException> { config.partitionValue }
+    // }
+
+    @Test
+    fun defaultPath() {
+        val user: User = createTestUser()
+        val config: SyncConfiguration = SyncConfiguration.with(user, setOf())
+        assertTrue(config.path.endsWith("/default.realm"), "Path is: ${config.path}")
+    }
+
+    // @Test
+    // fun initialSubscriptions() {
+    //     val user: User = createTestUser()
+    //     val handler: MutableSubscriptionSet.(realm: Realm) -> Unit = { /* Do nothing */ }
+    //     val config: SyncConfiguration = SyncConfiguration.Builder(user, setOf())
+    //         .initialSubscriptions(rerunOnOpen = true, handler)
+    //         .build()
+    //
+    //     assertEquals<InitialSubscriptionsCallback>(handler, config.initialSubscriptionsCallback!!)
+    //     assertTrue(config.rerunInitialSubscriptions)
+    // }
+
+    // @Test
+    // fun rerunInitialSubscriptions() {
+    //     val user: User = createTestUser()
+    //     val config1 = SyncConfiguration.with(user, setOf())
+    //     assertFalse(config1.rerunInitialSubscriptions)
+    //
+    //     val config2 = SyncConfiguration.Builder(user, setOf())
+    //         .initialSubscriptions(true) {
+    //             // Do nothing
+    //         }
+    //         .build()
+    //     assertTrue(config2.rerunInitialSubscriptions)
+    // }
+
+    // @Test
+    // fun defaultClientResetStrategy() {
+    //     val user: User = createTestUser(app)
+    //     val handler = SyncConfiguration.InitialFlexibleSyncSubscriptions { realm, subscriptions ->
+    //         // Do nothing
+    //     }
+    //     val config: SyncConfiguration = SyncConfiguration.defaultConfig(user)
+    //     assertTrue(config.syncClientResetStrategy is ManuallyRecoverUnsyncedChangesStrategy)
+    // }
+
+    @Test
+    fun overrideDefaultPath() {
+        val user: User = createTestUser()
+        val config: SyncConfiguration = SyncConfiguration.Builder(user, setOf())
+            .name("custom.realm")
+            .build()
```

Comment:
Because it conflicted with the `AppConfiguration.syncRoot` in weird ways. We discussed it back then and agreed that having `syncRoot` + `name` for Sync Realms and `directory() + name()` for normal Realms was what made the most sense.